### ISSUE TITLE
Add a lint for checking test file name

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/linter#3493]
**Describe the rule you'd like to see implemented**

For `dart test` to run tests, the test files need be be named "*_test.dart" (unless changed in the config file).  
The problem is, if someone makes a mistake in the file name, dart test will not execute it – yet it'll be difficult to catch.

I propose having a lint rule to fix this. This lint rule would detect that a file within `/test` contains a `main`. And if so, the file name should use `*_test.dart`

**Examples**

```dart
// test/foo.dart 
void main() {} // KO

// test/foo_test.dart
void main() {} // OK
```

**Comment by incendial** [comment]
This lint is now available in Dart Code Metrics https://dartcodemetrics.dev/docs/rules/common/prefer-correct-test-file-name
